### PR TITLE
CB-1201 :  Support for DB setup for Oozie, Ranger

### PR DESCRIPTION
**Pull request by smaniraju**
Added Support for Configuring DB Properties for Oozie, Ranger.

### CODE REVIEW

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/RangerConfigProvider.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        result.add(new ApiClusterTemplateConfig().name("ranger_database_type").variable("ranger-ranger_database_type"));
+        result.add(new ApiClusterTemplateConfig().name("ranger_database_user").variable("ranger-ranger_database_user"));
+        result.add(new ApiClusterTemplateConfig().name("ranger_database_user_password").variable("ranger-ranger_database_user_password"));
+        return result;
+    }
+
+    @Override
+    public List<ApiClusterTemplateVariable> getServiceConfigVariables(TemplatePreparationObject source) {
+        List<ApiClusterTemplateVariable> result = new ArrayList<>();
+        RdsView hiveView = new RdsView(getFirstRDSConfigOptional(source).get());
```
Comment (doktoric) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/RangerConfigProvider.java]: can you please rename this variable ?

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/SqoopConfigProvider.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        result.add(new ApiClusterTemplateConfig().name("sqoop_repository_database_type").variable("sqoop-sqoop_repository_database_type"));
+        result.add(new ApiClusterTemplateConfig().name("sqoop_repository_database_user").variable("sqoop-sqoop_repository_database_user"));
+        result.add(new ApiClusterTemplateConfig().name("sqoop_repository_database_password").variable("sqoop-sqoop_repository_database_password"));
+        return result;
+    }
+
+    @Override
+    public List<ApiClusterTemplateVariable> getServiceConfigVariables(TemplatePreparationObject source) {
+        List<ApiClusterTemplateVariable> result = new ArrayList<>();
+        RdsView oozieRdsView = new RdsView(getFirstRDSConfigOptional(source).get());
```
Comment (doktoric) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/SqoopConfigProvider.java]: can you please rename ?

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmDBConfigProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import com.cloudera.api.swagger.model.ApiClusterTemplateService;
+import com.sequenceiq.cloudbreak.util.FileReaderUtils;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CmDBConfigProviderTest {
+
+    private CmTemplateProcessor underTest;
+
+    @Test
+    public void testSqoopServiceDBConfigs() {
```
Comment (doktoric) [inline template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmDBConfigProviderTest.java]: Can you please separate this file and create one for envery service type ?

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/RangerConfigProvider.java (hunk 1 above)
Comment (smaniraju) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/RangerConfigProvider.java]: Renamed the variable

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/SqoopConfigProvider.java (hunk 2 above)
Comment (smaniraju) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/SqoopConfigProvider.java]: Renamed the variable.

File: template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmDBConfigProviderTest.java (hunk 3 above)
Comment (smaniraju) [reply template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmDBConfigProviderTest.java]: Sure, separated the test files into once for each service. Thanks

**Comment by doktoric** [comment]
please also add Sqoop to SupportedDatabaseProvider.java

**Comment by keyki** [comment]
I think for oozie and sqoop it's missing that we create the embedded DB without external DB. Take a look at `HiveRdsConfigProvider` and `RangerRdsConfigProvider`.

**Comment by smaniraju** [comment]
> I think for oozie and sqoop it's missing that we create the embedded DB without external DB. Take a look at `HiveRdsConfigProvider` and `RangerRdsConfigProvider`. For Ranger we also need to extend `isRdsConfigNeedForRangerAdmin` to support CM templates. Otherwise the PR looks good.

Added embedded DB Config providers for hive and ranger

**Comment by smaniraju** [comment]
> please also add Sqoop to SupportedDatabaseProvider.java

Currently not added support for Sqoop since it is still being discussed.

**Pull request merged**